contracts/cw20-streams/src/contract.rs

```diff
@@ -0,0 +1,446 @@
+        let end_time = mock_env().block.time.plus_seconds(20).seconds();
+
+        let msg = ExecuteMsg::Receive(Cw20ReceiveMsg {
+            sender,
+            amount,
+            msg: to_binary(&ReceiveMsg::CreateStream {
+                recipient,
+                start_time,
+                end_time,
+            })
+            .unwrap(),
+        });
+        info.sender = Addr::unchecked(MOCK_CONTRACT_ADDR);
+        let err = execute(deps.as_mut(), mock_env(), info, msg).unwrap_err();
+        match err {
+            ContractError::InvalidStartTime {} => {}
+            e => panic!("unexpected error: {}", e),
+        }
+    }
+
+    #[test]
+    fn invalid_cw20_addr() {
+        let mut deps = mock_dependencies();
+
+        let msg = InstantiateMsg {
+            owner: None,
+            cw20_addr: String::from(MOCK_CONTRACT_ADDR),
+        };
+        let mut info = mock_info("Alice", &[]);
+        instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        let sender = Addr::unchecked("Alice").to_string();
+        let recipient = Addr::unchecked("Bob").to_string();
+        let amount = Uint128::new(100);
+        let start_time = mock_env().block.time.plus_seconds(100).seconds();
+        let end_time = mock_env().block.time.plus_seconds(200).seconds();
+
+        let msg = ExecuteMsg::Receive(Cw20ReceiveMsg {
+            sender,
+            amount,
+            msg: to_binary(&ReceiveMsg::CreateStream {
+                recipient,
+                start_time,
+                end_time,
+            })
+            .unwrap(),
+        });
+        info.sender = Addr::unchecked("wrongCw20");
+        let err = execute(deps.as_mut(), mock_env(), info, msg).unwrap_err();
+
+        match err {
+            ContractError::Unauthorized {} => {}
+            e => panic!("unexpected error: {}", e),
+        }
+    }
+
+    #[test]
+    fn invalid_deposit_amount() {
+        let mut deps = mock_dependencies();
+
+        let msg = InstantiateMsg {
+            owner: None,
+            cw20_addr: String::from(MOCK_CONTRACT_ADDR),
+        };
+        let mut info = mock_info("Alice", &[]);
+        instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        let sender = Addr::unchecked("Alice").to_string();
+        let recipient = Addr::unchecked("Bob").to_string();
+        let amount = Uint128::new(3);
+        let start_time = mock_env().block.time.plus_seconds(100).seconds();
+        let end_time = mock_env().block.time.plus_seconds(200).seconds();
+
+        let msg = ExecuteMsg::Receive(Cw20ReceiveMsg {
+            sender,
+            amount,
+            msg: to_binary(&ReceiveMsg::CreateStream {
+                recipient,
+                start_time,
+                end_time,
+            })
+            .unwrap(),
+        });
+        info.sender = Addr::unchecked(MOCK_CONTRACT_ADDR);
+        execute(deps.as_mut(), mock_env(), info, msg).unwrap_err();
```

Comment:
Let's also assert which error it is. This makes it more clear which error case we trigger.

```rust
let err = execute(deps.as_mut(), mock_env(), info, msg).unwrap_err();
assert_eq!(err, ContractError::InvalidDuration{});
```

Reply:
Done. Having all the other test cases use this pattern as well. I like the 1 line assert